examples/ecn_alan.ipynb

```diff
@@ -0,0 +1,199 @@
+   },
+   "source": [
+    "## Building an ECN Eval\n",
+    "\n",
+    "This notebook shows how to build and run an eval following instructions in docs/build-eval.md"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "\n",
+    "\n",
+    "# Assuming this notebook is in examples/\n",
+    "registry_path = os.path.join(os.getcwd(), \"../evals/registry\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "import datasets\n",
+    "\n",
+    "HF_TOKEN = \"...\"\n",
+    "dataset = datasets.load_dataset(\"Alan-health/ecn\", token=HF_TOKEN)"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n",
+     "is_executing": true
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "\n",
+    "sys_msg = \"Vous êtes un expert médical. Vous devez répondre à des questions médicales à choix multiple. Choisissez les réponses correctes parmi les réponses proposées, indiquées par les lettres A, B, C, D ou E. Si la bonne réponse est la réponse A, répondez simplement 'A'. Les questions peuvents avoir plusieurs réponses. Dans ce cas, répondez avec toutes les réponses, séparées par une virgule, par example 'B,D,E'.\"\n",
+    "\n",
+    "\n",
+    "def create_chat_prompt(sample):\n",
+    "    user_prompt = (\n",
+    "        f\"{sample['question']}\\n\"\n",
+    "        f\"A: {sample['answer_A']}\\n\"\n",
+    "        f\"B: {sample['answer_B']}\\n\"\n",
+    "        f\"C: {sample['answer_C']}\\n\"\n",
+    "        f\"D: {sample['answer_D']}\\n\"\n",
+    "        f\"E: {sample['answer_E']}\\n\"\n",
+    "        f\"Réponse:\"\n",
+    "    )\n",
+    "    return [\n",
+    "        {\"role\": \"system\", \"content\": sys_msg},\n",
+    "        {\"role\": \"user\", \"content\": user_prompt}\n",
+    "    ]\n",
+    "\n",
+    "def create_chat_example(sample):\n",
+    "    \"\"\"\n",
+    "    Form few-shot prompts in the recommended format: https://github.com/openai/openai-python/blob/main/chatml.md#few-shot-prompting\n",
+    "    \"\"\"\n",
+    "    user_prompt = (\n",
+    "        f\"{sample['question']}\\n\"\n",
+    "        f\"A: {sample['answer_A']}\\n\"\n",
+    "        f\"B: {sample['answer_B']}\\n\"\n",
+    "        f\"C: {sample['answer_C']}\\n\"\n",
+    "        f\"D: {sample['answer_D']}\\n\"\n",
+    "        f\"E: {sample['answer_E']}\\n\"\n",
+    "        f\"Réponse:\"\n",
+    "    )\n",
+    "    return [\n",
+    "        {\"role\": \"system\", \"content\": user_prompt, \"name\": \"example_user\"},\n",
```

Comment:
I had seen that! Super weird, new and curious how they actually use it.

Reply:
are you referring to the 'name' field? Not sure either how it's used.

Btw, it's interesting the few-shots are in the system prompt. Not sure how much that matters.